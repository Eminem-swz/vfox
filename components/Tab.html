<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Tab 标签列表 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.992fff1b.css">
    <link rel="modulepreload" href="/vfox/assets/Home.3f173d00.js">
    <link rel="modulepreload" href="/vfox/assets/app.2d0a5a59.js">
    <link rel="modulepreload" href="/vfox/assets/components_Tab.md.c648ef6d.lean.js">
    <link rel="modulepreload" href="/vfox/assets/app.2d0a5a59.js">
    <meta name="twitter:title" content="Tab 标签列表 | Vfox for Vue3">
    <meta property="og:title" content="Tab 标签列表 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-675d8756><div class="sidebar-button" data-v-675d8756><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-675d8756 data-v-4a583abe><!----> Vfox for Vue3</a><div class="flex-grow" data-v-675d8756></div><div class="nav" data-v-675d8756><nav class="nav-links" data-v-675d8756 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/apis/LocalStorage" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-83e92a68><nav class="nav-links nav" data-v-83e92a68 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/apis/LocalStorage" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-83e92a68><!--[--><li class="sidebar-link"><p class="sidebar-link-item">基础</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Button">Button</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Icon">Icon</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Image">Image</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Drawer">Drawer</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Modal">Modal</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dropdown">Dropdown</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">反馈</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dialog">Dialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Notify">Notify</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SwipeCell">SwipeCell</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">数据输入</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Form">Form</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Input">Input</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Radio">Radio</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Checkbox">Checkbox</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Slider">Slider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Switch">Switch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stepper">Stepper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Calendar">Calendar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SearchBar">SearchBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NumberKeyboard">NumberKeyboard</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImageUploader">ImageUploader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Range">Range</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">导航</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/BackTop">BackTop</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NavBar">NavBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopMenu">PopMenu</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/components/Tab">Tab</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#props">Props</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#options-的结构">options 的结构</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#css">CSS</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#events">Events</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#change-的回调参数">change 的回调参数</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#methods">Methods</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SideTab">SideTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabBar">TabBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabView">TabView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Sticky">Sticky</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollTab">ScrollTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/IndexView">IndexView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Fixed">Fixed</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Pagination">Pagination</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">展示</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Layout">Layout</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cell">Cell</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Group">Group</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Collapse">Collapse</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Empty">Empty</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Skeleton">Skeleton</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Swiper">Swiper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImagePreview">ImagePreview</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/FlatList">FlatList</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NoticeBar">NoticeBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Divider">Divider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Badge">Badge</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Progress">Progress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tag">Tag</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Order">Order</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountDown">CountDown</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Steps">Steps</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Price">Price</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Avatar">Avatar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Timeline">Timeline</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stopwatch">Stopwatch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Result">Result</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/LoadMore">LoadMore</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TimeAgo">TimeAgo</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountUp">CountUp</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CircleProgress">CircleProgress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActivityIndicator">ActivityIndicator</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">其他</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Copy">Copy</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><!-- TODO: make this button accessible --><div class="sidebar-mask"></div><main class="page" data-v-7eddb2c4><div class="container" data-v-7eddb2c4><!--[--><!--[--><!----><!--]--><!--[--><div><iframe style="width:375px;height:667px;margin-top:1.5rem;" src="https://godxiaoji.github.io/vfox/demo/#/Tab" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-7eddb2c4><div><h1 id="tab-标签列表" tabindex="-1">Tab 标签列表 <a class="header-anchor" href="#tab-标签列表" aria-hidden="true">#</a></h1><h2 id="props" tabindex="-1">Props <a class="header-anchor" href="#props" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>options</td><td>object[]/string[]/number[]</td><td>[]</td><td>是</td><td>tab 数据集</td></tr><tr><td>v-model:active-value</td><td>string, number</td><td></td><td>否</td><td>当前激活项的 value 值</td></tr><tr><td>scroll-threshold</td><td>number</td><td>4</td><td>否</td><td>超过 <code>scrollThreshold</code> 个 Tab 使用滚动形式</td></tr><tr><td>field-names</td><td>object</td><td>{ label: &#39;label&#39;, value: &#39;value&#39; }</td><td>否</td><td>自定义 options 中 label value 的字段 key</td></tr></tbody></table><h3 id="options-的结构" tabindex="-1">options 的结构 <a class="header-anchor" href="#options-的结构" aria-hidden="true">#</a></h3><table><thead><tr><th>key</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>value</td><td>string/number</td><td></td><td>是</td><td>唯一值（v-model:active-value 使用）</td></tr><tr><td>label</td><td>string</td><td></td><td>是</td><td>标签名</td></tr><tr><td>icon</td><td>string/Component</td><td></td><td>否</td><td>设置图标，使用 <a href="./Icon.html">Icon</a> 组件</td></tr><tr><td>activeIcon</td><td>string/Component</td><td></td><td>否</td><td>设置激活态图标，没有设置则沿用 <code>icon</code> 属性，使用 <a href="./Icon.html">Icon</a> 组件</td></tr><tr><td>badge</td><td>string/number/object</td><td></td><td>否</td><td>徽标，使用 <a href="./Badge.html">Badge</a> 组件，可传入一个基于 Badge props 的对象</td></tr><tr><td>subLabel</td><td>string</td><td></td><td>否</td><td>副标签名，如果设置了该字段，则显示带有副标签的列表</td></tr></tbody></table><div class="language-"><pre><code>[
  {
    value: 1,
    label: &#39;首页&#39;,
    icon: &#39;HomeOutlined&#39;,
    badge: &#39;热&#39;
  },
  {
    value: 2,
    label: &#39;搜索&#39;,
    icon: &#39;SearchOutlined&#39;,
    badge: {
      dot: true,
      content: 1
    }
  }
]
</code></pre></div><p>也可以直接设置为 <code>string[]</code> 或 <code>number[]</code>，如：</p><div class="language-"><pre><code>[&#39;aaa&#39;, &#39;bbb&#39;, &#39;ccc&#39;]
</code></pre></div><p>将被转为：</p><div class="language-"><pre><code>[
  {
    value: &#39;aaa&#39;,
    label: &#39;aaa&#39;
  },
  {
    value: &#39;bbb&#39;,
    label: &#39;bbb&#39;
  }
  {
    value: &#39;ccc&#39;,
    label: &#39;ccc&#39;
  }
]
</code></pre></div><p>注：注意数组项要保持唯一性。</p><h2 id="css" tabindex="-1">CSS <a class="header-anchor" href="#css" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>默认值</th><th>说明</th></tr></thead><tbody><tr><td>--tab-active-color</td><td>#262626</td><td>激活色</td></tr><tr><td>--tab-color</td><td>#595959</td><td>默认色</td></tr></tbody></table><h2 id="events" tabindex="-1">Events <a class="header-anchor" href="#events" aria-hidden="true">#</a></h2><table><thead><tr><th>事件</th><th>描述</th><th>回调函数参数</th></tr></thead><tbody><tr><td>change</td><td>点击切换 tab 时触发</td><td>{ value, index }</td></tr></tbody></table><h3 id="change-的回调参数" tabindex="-1">change 的回调参数 <a class="header-anchor" href="#change-的回调参数" aria-hidden="true">#</a></h3><table><thead><tr><th>参数</th><th>类型</th><th>描述</th></tr></thead><tbody><tr><td>value</td><td>string/number</td><td>当前激活项的 value 值</td></tr><tr><td>index</td><td>number</td><td>当前激活项的索引值</td></tr></tbody></table><h2 id="methods" tabindex="-1">Methods <a class="header-anchor" href="#methods" aria-hidden="true">#</a></h2><table><thead><tr><th>方法名</th><th>说明</th><th>参数</th></tr></thead><tbody><tr><td>switchTo</td><td>切换到指定 Tab</td><td>(value: string/number) =&gt; void</td></tr><tr><td>switchToIndex</td><td>切换到指定索引的 Tab</td><td>(index: number) =&gt; void</td></tr></tbody></table></div></div><footer class="page-footer" data-v-7eddb2c4 data-v-fb8d84c6><div class="edit" data-v-fb8d84c6><div class="edit-link" data-v-fb8d84c6 data-v-1ed99556><!----></div></div><div class="updated" data-v-fb8d84c6><!----></div></footer><div class="next-and-prev-link" data-v-7eddb2c4 data-v-38ede35f><div class="container" data-v-38ede35f><div class="prev" data-v-38ede35f><a class="link" href="/vfox/components/ScrollView" data-v-38ede35f><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-38ede35f><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-38ede35f>ScrollView</span></a></div><div class="next" data-v-38ede35f><a class="link" href="/vfox/components/SideTab" data-v-38ede35f><span class="text" data-v-38ede35f>SideTab</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-38ede35f><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"apis_drawer.md\":\"807c95b6\",\"apis_intersectionobserver.md\":\"e37c35bc\",\"apis_localstorage.md\":\"2aad00b8\",\"apis_modal.md\":\"6ee8fade\",\"apis_notify.md\":\"8bfa36e9\",\"apis_popover.md\":\"0e69f7d9\",\"apis_preview.md\":\"3573010e\",\"apis_scroll.md\":\"bf4fbb81\",\"apis_selectorquery.md\":\"eeb4fa3e\",\"apis_toast.md\":\"6e58cf0f\",\"apis_index.md\":\"fa3c10d9\",\"components_actionsheet.md\":\"5649d6d5\",\"components_activityindicator.md\":\"822760d5\",\"components_avatar.md\":\"322719e6\",\"components_backtop.md\":\"8f62f9c5\",\"components_badge.md\":\"dfb66bcc\",\"components_button.md\":\"44647074\",\"components_calendar.md\":\"20a07da0\",\"components_cascader.md\":\"eb627590\",\"components_cell.md\":\"7038b7a5\",\"components_checkbox.md\":\"d573f332\",\"components_circleprogress.md\":\"1ea13341\",\"components_collapse.md\":\"5517897a\",\"components_copy.md\":\"c4bab33c\",\"components_countdown.md\":\"2929e945\",\"components_countup.md\":\"f9b2d192\",\"components_datepicker.md\":\"207bded6\",\"components_dialog.md\":\"06366e8d\",\"components_divider.md\":\"3b946731\",\"components_drawer.md\":\"b4667761\",\"components_dropdown.md\":\"4dacef3f\",\"components_empty.md\":\"ada473c0\",\"components_fixed.md\":\"c4e71d40\",\"components_flatlist.md\":\"afa889a4\",\"components_form.md\":\"2479956c\",\"components_group.md\":\"c4af0f4a\",\"components_icon.md\":\"6ba6c240\",\"components_image.md\":\"66ae6e8c\",\"components_imagepreview.md\":\"71551303\",\"components_imageuploader.md\":\"75c03425\",\"components_indexview.md\":\"1a5bd160\",\"components_input.md\":\"97fbd548\",\"components_layout.md\":\"355e4553\",\"components_loadmore.md\":\"fe15435c\",\"components_modal.md\":\"c273801d\",\"components_navbar.md\":\"5762eeed\",\"components_noticebar.md\":\"b3909d3a\",\"components_notify.md\":\"53dbf91a\",\"components_numberkeyboard.md\":\"8ef6ae94\",\"components_order.md\":\"6696d4b2\",\"components_pagination.md\":\"9c826c94\",\"components_picker.md\":\"c80dceb0\",\"components_popdialog.md\":\"063a5aca\",\"components_popmenu.md\":\"e1097abc\",\"components_popover.md\":\"9f5b0a02\",\"components_price.md\":\"492eae3c\",\"components_progress.md\":\"bf71d393\",\"components_radio.md\":\"b42f6542\",\"components_range.md\":\"ac67719e\",\"components_rate.md\":\"4304ffec\",\"components_result.md\":\"bca6d007\",\"components_scrolltab.md\":\"4195a3a6\",\"components_scrollview.md\":\"61c2476f\",\"components_searchbar.md\":\"428cd164\",\"components_sidetab.md\":\"66b5e724\",\"components_skeleton.md\":\"c06ae4f6\",\"components_slider.md\":\"c6802d3e\",\"components_stepper.md\":\"b9ffff56\",\"components_steps.md\":\"0f722099\",\"components_sticky.md\":\"4af5d44d\",\"components_stopwatch.md\":\"879961a8\",\"components_swipecell.md\":\"004fda58\",\"components_swiper.md\":\"09e0d987\",\"components_switch.md\":\"9e773d64\",\"components_tab.md\":\"c648ef6d\",\"components_tabbar.md\":\"80352fdf\",\"components_tabview.md\":\"203f9741\",\"components_tag.md\":\"00ff84ab\",\"components_timeago.md\":\"9c152701\",\"components_timeline.md\":\"1a4f3a94\",\"components_toast.md\":\"a6097551\",\"components_index.md\":\"bdc99e52\",\"design_index.md\":\"44b4149d\",\"index.md\":\"4649973d\"}")</script>
    <script type="module" async src="/vfox/assets/app.2d0a5a59.js"></script>
    
  </body>
</html>