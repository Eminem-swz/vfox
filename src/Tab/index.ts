import { App } from 'vue'
import { SFCWithInstall } from '../utils/types'
import Tab from './Tab.vue'

const _Tab: SFCWithInstall<typeof Tab> = Object.assign(Tab, {
  install: function(app: App) {
    app.component(Tab.name, Tab)
  }
})

export default _Tab
