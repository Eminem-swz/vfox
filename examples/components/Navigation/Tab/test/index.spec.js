/* global describe,test,expect */

import { shallowMount } from '@vue/test-utils'
import Tab from '@/Tab'

import { mixTabList, tabList } from '../data'

describe('Tab', () => {
  test('should render set options correctly', () => {
    const wrapper = shallowMount(Tab, {
      props: {
        options: tabList
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render set mix options correctly', () => {
    const wrapper = shallowMount(Tab, {
      props: {
        options: mixTabList
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
