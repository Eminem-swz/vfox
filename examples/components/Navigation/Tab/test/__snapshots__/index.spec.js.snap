// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should render set mix options correctly 1`] = `
<div class="fx-tab no--scroll">
  <ul class="fx-tab_list">
    <li class="fx-tab_item fx-vertical-hairline active">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline active-next">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
  </ul>
</div>
`;

exports[`Tab should render set options correctly 1`] = `
<div class="fx-tab">
  <ul class="fx-tab_list">
    <li class="fx-tab_item fx-vertical-hairline active">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline active-next">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
    <li class="fx-tab_item fx-vertical-hairline">
      <fx-badge-stub class="fx-tab_item-inner"></fx-badge-stub>
      <!--v-if-->
    </li>
  </ul>
</div>
`;
